__test__/components/ReloadPage.test.tsx

```diff
@@ -0,0 +1,186 @@
+  }
+}];
+
+const mockedSetSemesters = jest.fn();
+const mockedHandleReturn = jest.fn();
+
+test("Reload Button is Visible", async () => {
+  const reloadButton = render(<ReloadPage
+  scheduleData={mockedSchduleData}
+  sems={mockedSem}
+  requirementsData={mockedRequirementsDisplayList}
+  setSemesters={mockedSetSemesters}
+  handleReturn={mockedHandleReturn}
+  />);
+  expect(reloadButton.baseElement).toBeInTheDocument();
+});
+
+// Functional test
+test("Reload Button is Functional", async () => {
+  const user = setupUser();
+  const reqs: RequirementComponentType[] = await fetchApiJson(
+    "/api/requirements?conid=14"
+  );
+  // update database reqs to the req type we use
+  reqs.forEach((req) => { req.courseCountTaken = 0; req.coursesTaken = ""; req.creditCountTaken = 0; req.percentage = 0; });
+  const coursesMajor: CourseType[] = await fetchApiJson(
+    "/api/courses/major?majid=2"
+  );
+
+  // Get general requirements
+  const reqsGen: RequirementComponentType[] = await fetchApiJson(
+    "/api/requirements/gen"
+  );
+  // update database reqs to the req type we use
+  reqs.forEach((req) => { req.courseCountTaken = 0; req.coursesTaken = ""; req.creditCountTaken = 0; req.percentage = 0; });
+  // get gen ed courses
+  const courses: CourseType[] = await fetchApiJson(
+    "/api/courses/geneds"
+  );
+  const bar = render(<FourYearPlanPage PassedCourseList={coursesMajor.concat(courses)} requirements={reqs} requirementsGen={reqsGen} />);
+  expect(bar.baseElement).toMatchSnapshot();
+
+  const firstYear = parentEl(screen.getByText(/Year 1/i), "MuiAccordion");
+  expect(firstYear).toBeInTheDocument();
+  const fallSemester = parentEl(
+    within(firstYear).getByText(/Fall/i),
+    "Semester"
+  );
+  expect(fallSemester).toBeInTheDocument();
+
+  const categoryTab = screen.getByTestId("category-tab");
+  await user.click(categoryTab);
+  await user.selectAutocomplete("Course Category", "CS - Computer Science Core");
+
+  const rb = screen.getByTestId("rloadButton");
+
+  // creating the course and drags it to the semester
+  const course = screen.getAllByTestId("course");
+  fireEvent.dragStart(course[0]);
+  fireEvent.dragEnter(fallSemester);
+  fireEvent.dragOver(fallSemester);
+  fireEvent.drop(fallSemester);
+
+  // checking the course gets added
+  const openButton = screen.getByTestId("openDrawer");
+  await user.click(openButton);
+  const reqPercentages = screen.getAllByTestId("reqPercent");
+  expect(reqPercentages[0]).toContainHTML("3.3");
+  expect(reqPercentages[2]).toContainHTML("9.1");
+  const closeButton = screen.getByTestId("closeDrawer");
+  await user.click(closeButton);
+
+  // refreshes the page and resets the credits
+  await user.click(rb);
+  await user.click(openButton);
+  expect(reqPercentages[0]).toContainHTML("0");
+  expect(reqPercentages[2]).toContainHTML("0");
+  await user.click(closeButton);
+
+  // add 2 courses then retry
+  fireEvent.dragStart(course[0]);
```

Comment:
use the function exposed by drag-drop.ts to implement drag and drop

Reply:
Fixed